src/js/view/captionsrenderer.js

```diff
@@ -192,7 +193,57 @@ define([
             this.populate(_model.get('captionsTrack'));
         };
 
-        function addEdgeStyle(option, style, fontOpacity) {
+        function setupShadowDOMStyles(playerElement, windowStyle, textStyle) {
+            // We have to use a style element to apply captions styles to user-agent Shadow DOM nodes.
+
+            // Caption window styles
+            var windowStyleText = '#' + playerElement.id + ' .jw-video::-webkit-media-text-track-display {';
+            if (windowStyle.backgroundColor) {
+                windowStyleText += ' background-color: ' + windowStyle.backgroundColor + ';';
+            }
+            windowStyleText += '}';
+
+            // Caption text styles
+            // The cue element is a div in Chrome
+            var textStyleText = '#' + playerElement.id + ' .jw-video::cue {';
+
+            // In Safari, the cue element is a span and has a parent backdrop div
+            var backdropStyleText =  '#' + playerElement.id + ' .jw-video::-webkit-media-text-track-display-backdrop {';
+
+            if (textStyle.color) {
+                textStyleText += ' color: ' + textStyle.color + ';';
+            }
+
+            if (textStyle.fontFamily) {
+                textStyleText += ' font-family: "' + textStyle.fontFamily + '";';
+            }
+
+            if (textStyle.fontStyle) {
+                textStyleText += ' font-style: ' + textStyle.fontStyle + ';';
+            }
+
+            if (textStyle.fontWeight) {
+                textStyleText += ' font-weight: ' + textStyle.fontWeight + ';';
+            }
+
+            if (textStyle.backgroundColor) {
+                textStyleText += ' background-color: ' + textStyle.backgroundColor + ';';
+                // Style text background in Safari. Needs to be !important to override browser style
+                backdropStyleText += ' background-color: ' + textStyle.backgroundColor + ' !important;';
+            } else if (textStyle.textShadow) {
+                textStyleText += ' text-shadow: ' + textStyle.textShadow + ';';
+            }
+            textStyleText += '}';
+            backdropStyleText += '}';
+
+
+
+            var style = document.createElement('style');
+            style.textContent = windowStyleText + textStyleText + backdropStyleText;
+            playerElement.appendChild(style);
```

Comment:
We have a util for this "utils.css":

```js
cssUtils.css('#' + playerElement.id + ' .jw-video::-webkit-media-text-track-display', styleObject);
```

Don't construct the style text yourself. This util does that for you.